addons/test_mail/tests/test_discuss.py

```diff
@@ -130,3 +130,17 @@ def test_mail_cc_recipient_suggestion(self):
             (False, 'cc1@example.com', 'CC Email'),
             (False, 'cc2@example.com', 'CC Email'),
         ], 'cc should be in suggestions')
+
+    def test_unlink_notification_message(self):
+        channel = self.env['mail.channel'].with_user(self.user_employee).create({'name': 'testChannel'})
```

Comment:
You may use the @users decorator for the method (and do the message search as sudo). It would be more explicit in my opinion, as you prefer :) .

Reply:
Yup I agree on this as I used `.with_user(self.user_employee)` few times. It'll be more explicit and easier to read :)